src/io/index.js

```diff
@@ -0,0 +1,303 @@
+import fs from '../lib/fs';
+var sh = new fs.Shell();
+import formatFS from '../lib/format-fs';
+import buffer from '../lib/buffer';
+import pth from '../lib/path';
+
+import { getMimeType } from './content-type';
+import { formatDir } from './html-formatter';
+import registerRoutes from './routes';
+
+/**
+ *
+ */
+export default class {
+    init(workbox) {
+        registerRoutes(workbox, this);
+    }
+
+    // Creates a file from an encoded text
+    async createFileFromEncodedText(path) {
+        return new Promise((resolve, reject) => {
+            var text = pth.basename(path);
+            var paths = path.substring(0, path.indexOf(text) - 1);
+            var fileName = pth.basename(paths);
+            var filePath = pth.join(pth.dirname(paths), fileName);
+            
+            fs.writeFile(filePath, text, function(err) {
+                if (err) reject({ success: false, err: err });
+                resolve({ success: true, path: pth.dirname(paths)});
+            });
+        });
+    }
+
+    // Converts Data URL to a Blob
+    dataURLtoBlob(dataurl) {
+        var arr = dataurl.split(','),
+            mime = arr[0].match(/:(.*?);/)[1],
+            bstr = atob(arr[1]),
+            n = bstr.length,
+            u8arr = new Uint8Array(n);
+        while (n--) {
+            u8arr[n] = bstr.charCodeAt(n);
+        }
+        return new Blob([u8arr], { type: mime });
+    }
+
+    // Makes FileReader Api work with async/await
+    // Get file as ArrayBuffer
+    async readUploadedFileAsArrayBuffer(inputFile) {
+        const temporaryFileReader = new FileReader();
+        return new Promise((resolve, reject) => {
+            temporaryFileReader.onerror = () => {
+                temporaryFileReader.abort();
+                reject(new DOMException('Problem parsing input file.'));
+            };
+
+            temporaryFileReader.onload = () => {
+                resolve(temporaryFileReader.result);
+            };
+            temporaryFileReader.readAsArrayBuffer(inputFile);
+        });
+    }
+
+    // Makes FileReader Api work with async/await
+    // Get file as DataURL
+    async readUploadedFileAsDataURL(inputFile) {
+        const temporaryFileReader = new FileReader();
+        return new Promise((resolve, reject) => {
+            temporaryFileReader.onerror = () => {
+                temporaryFileReader.abort();
+                reject(new DOMException('Problem parsing input file.'));
```

Comment:
On [this line](https://github.com/humphd/next/pull/93/files#diff-ae96a6677215bee1532bd7d6326e2ab7R101) you are using text exceptions, but here it is `DOMException`. Can you stick with only one style of exception, please?